mdsuite/utils/testing.py

```diff
@@ -58,6 +61,40 @@ def assertDeepAlmostEqual(expected, actual, *args, **kwargs):
         assert expected == actual
 
 
+class MDSuiteProcess(multiprocessing.Process):
+    """
+    Process class for use in ZnVis testing.
+    """
```

Comment:
Can you explain this class and the associated tests to me? I don't really understand what it does tbh.

Reply:
Absolutely. If there is a better way to handle this also please let me know. 

When we want to run a test calling Open3D we need to open it. Open3D will run a separate process as it allows for callbacks but this means that it cannot be forcefully closed during it's run, it must be closed by a user. The idea behind using multiprocessing for a test is that we can open the code in a new process, let it run and then use `process.terminate()` to forcefully end it. The reason the Process is sub-classed is because we want to use it for a test and therefore need a way of collecting the exceptions or errors that were raised on the deployed process. This class collects these methods such that they can be looked at in the test. Without this functionality, if an exception is raised on the deployed process all of the tests will still pass.